src/vit-testing/mainnet-tools/src/bin/fake_snapshot_tool.rs

```diff
@@ -0,0 +1,47 @@
+use clap::Parser;
```

Comment:
I'm not sure we want a separate binary for the tool with a mock data source. We don't duplicate too much, since most of the logic is in the voting tools `lib.rs`, but IMO it's cleaner and "closer to production" if we just have a CLI arg like `--mock-json-path` or something that is mutually exclusive with a set of real DB credentials

So I'm thinking a rough structure like this:
```rust
trait Db {
  // queries
}

struct RealDb { ... }
impl Db for RealDb { ... }

struct MockDb { ... }  // currently called `DbSyncInstance` IIRC
impl Db for MockDb {}

enum DbArgs {
  Real {
    host: DbHost,
    // ...
  },
  Mock {
    json_path: PathBuf,
  }
}

fn get_db(args: &DbArgs) -> Result<Box<dyn Db>> {
  // build the correct database here
}
```
I haven't finished looking at everything but this seems like a slightly cleaner approach. But there might be a reason this doesn't work, not sure :shrug:

Reply:
Happy that you proposed this, as i was uncertain if adding this approach won't be too close to production and can be misused, but i agree with additional parameter with meaningful name should solve the problem